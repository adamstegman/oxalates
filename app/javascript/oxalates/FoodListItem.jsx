import React from 'react';
import PropTypes from 'prop-types';

import { cssify } from './cssify';
import { foodPropType } from './foodPropType';
import { listPropType } from './listPropType';

import './FoodListItem.scss';

export class FoodListItem extends React.Component {
  render() {
    const {
      food,
      list,
    } = this.props;

    const listClassName = cssify(list.name);
    return (
      <li className={`food-list-item ${listClassName}`}>
        {food.name} - {food.oxalateMg.toFixed(2)}mg - {food.serving}
      </li>
    );
  }
};

FoodListItem.propTypes = {
  food: foodPropType.isRequired,
  list: listPropType.isRequired,
};
