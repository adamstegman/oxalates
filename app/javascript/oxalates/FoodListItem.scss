@import "./lists";

$food_list_item_padding: 0.75em;
$food_list_item_font_size: 16px;
$destroy_food_list_item_icon_height: 22px;
$destroy_food_list_item_icon_width: 22px;
$destroy_food_list_item_link_height: 44px;
$destroy_food_list_item_link_width: 44px;

.food-list-item {
  padding: $food_list_item_padding;

  font-size: $food_list_item_font_size;
}

.destroy-food-list-item {
  width: $destroy_food_list_item_icon_width;
  height: $destroy_food_list_item_icon_height;
  padding: #{($destroy_food_list_item_link_height - $destroy_food_list_item_icon_height) / 2}
           #{($destroy_food_list_item_link_width  - $destroy_food_list_item_icon_width)  / 2};
}

.edit-food-list-item {
  @extend .food-list-item;

  display: block;
  -webkit-flex-grow: 2;
  flex-grow: 2;

  text-decoration: none;
}

.food-list-item-name {
  display: none;
}
input.food-list-item-name {
  @extend .food-list-item;

  width: 100%;

  -webkit-appearance: none;
}
