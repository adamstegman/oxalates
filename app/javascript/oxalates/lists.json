[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
