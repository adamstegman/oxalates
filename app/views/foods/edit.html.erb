<%= form_for @food do |f| %>
  <%= f.label :name %>
  <%= f.text_field :name %>
  <%= f.label :list %>
  <%= f.select :list, options_from_collection_for_select(all_lists, :id, :name, @food.list_id) %>

  <%= f.submit "Save" %>
<% end %>
