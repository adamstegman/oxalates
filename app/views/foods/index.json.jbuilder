json.foods(@foods) do |food|
  json.(food, :id, :name, :oxalate_mg, :serving)
end
