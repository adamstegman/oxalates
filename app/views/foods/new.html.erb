<%= form_for @food, html: {class: 'edit-food'} do |f| %>
  <p><%= f.label :name, "New food name" %></p>
  <p><%= f.text_field :name %></p>
  <p><%= f.label :oxalate_mg, "Oxalates per serving (mg)" %></p>
  <p><%= f.number_field :oxalate_mg, step: "any" %></p>
  <p><%= f.label :serving %></p>
  <p><%= f.text_field :serving %></p>

  <p><%= f.submit "Add" %></p>
<% end %>
