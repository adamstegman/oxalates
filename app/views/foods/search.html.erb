<h1>Results</h1>

<ul>
	<% @results.each do |r| %>
		<li><%= r.name + ": " + r.list.name %></li> 
	<% end %>
</ul>