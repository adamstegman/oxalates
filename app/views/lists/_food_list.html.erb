<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
