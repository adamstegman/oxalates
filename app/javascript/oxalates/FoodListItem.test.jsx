import React from 'react';
import renderer from 'react-test-renderer';

import foods from './__mocks__/foods.json';
import { FoodListItem } from './FoodListItem';
import lists from './__mocks__/lists.json';

const food = foods[0];
const list = lists[0];

test('FoodListItem renders a food', () => {
  const component = renderer.create(
    <FoodListItem food={food} list={list} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
