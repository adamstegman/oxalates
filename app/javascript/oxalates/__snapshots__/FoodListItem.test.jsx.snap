// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodListItem renders a food 1`] = `
<li
  className="food-list-item all"
>
  beetroot, steamed
   - 
  50.00
  mg - 
  just under 1/2 cup
</li>
`;
